Bottom: d56cf047a62e1d834f93eec581ff861efff6b169
Top:    554e6b222e782e46b98a3cd6e93fd0556bdee98b
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-01 15:37:45 +0800

autofs-5.0.7 - fix map read fail incorrectly set on master re-read

When re-reading the master map read failures for plus included maps
should not cause the map read to fail, they should be ingored.


---

diff --git a/CHANGELOG b/CHANGELOG
index 39388a5..b13a5f5 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -28,6 +28,7 @@
 - make yellow pages support optional.
 - modules/replicated.c: use sin6_addr.s6_addr32.
 - workaround missing GNU versionsort extension.
+- fix map read fail incorrectly set on master re-read.
 
 25/07/2012 autofs-5.0.7
 =======================
diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index facb305..411f280 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -397,8 +397,9 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 	unsigned int path_len, ent_len;
 	int entry, cur_state;
 
+	/* Don't return fail on self include, skip source */
 	if (master->recurse)
-		return NSS_STATUS_UNAVAIL;
+		return NSS_STATUS_SUCCESS;
 
 	if (master->depth > MAX_INCLUDE_DEPTH) {
 		error(logopt, MODPREFIX
@@ -443,7 +444,7 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 
 			inc = check_master_self_include(master, ctxt);
 			if (inc) 
-				master->recurse = 1;;
+				master->recurse = 1;
 			master->depth++;
 			status = lookup_nss_read_master(master, age);
 			if (!status) {
